test/services/indexer/repo/agencyJsonStream.test.js

```diff
@@ -0,0 +1,131 @@
+const fs = require('fs');
+const path = require('path');
+
+const chai = require('chai');
+const JsonStream = require('JSONStream');
+const JsonFile = require('jsonfile');
+const should = chai.should();
+
+const AgencyJsonStream = require('../../../../services/indexer/repo/AgencyJsonStream');
+
+describe('AgencyJsonStream', function() {
+  let agencyJsonStream;
+  let testDataDir;
+  let fallbackDataDir;
+  let fetchDataDir;
+  let agency;
+
+  beforeEach(function() {
+    testDataDir = path.join(__dirname, '/test_data');
+    fallbackDataDir = path.join(testDataDir, '/fallback');
+    fetchDataDir = path.join(testDataDir, '/fetched');
+    agency = JsonFile.readFileSync(path.join(testDataDir, 'test_agency_endpoints.json'))
+    agencyJsonStream = new AgencyJsonStream(fetchDataDir, fallbackDataDir);
+  });
+
+  it('Should save codeJson to disk', function(done) {
+    const codeJson = JsonFile.readFileSync(path.join(fallbackDataDir, '/FAKE.json'));
+
+    agencyJsonStream._saveFetchedCodeJson('FAKE', codeJson)
+      .then(data =>{
+        const fetchedCodeJson = JsonFile.readFileSync(path.join(fetchDataDir, '/FAKE.json'));
+        data.should.be.deep.equal(fetchedCodeJson);
+      })
+      .then(done, done);
+  });
+
+  it('Should return codejson from disk', function(done) {
+    const codeJson = JsonFile.readFileSync(path.join(fallbackDataDir, '/FAKE.json'));
+    agencyJsonStream._getAgencyCodeJson(agency[0])
+      .then(data => {
+        data.should.be.deep.equal(codeJson);
+      })
+      .then(done, done);
+  });
+
+  it('Should return array of repos', function(done){
+    const codeJson = JsonFile.readFileSync(path.join(fallbackDataDir, '/FAKE.json'));
+    const expectedRepos = codeJson.projects;
+    agencyJsonStream._validateAgencyRepos(agency[0], codeJson)
+      .then(repos => {
+        repos.should.be.deep.equal(expectedRepos);
+      })
+      .then(done, done);
+  });
+
+  it('Should return fomatted repos', function(done){
+    const expectedRepo = {
```

Comment:
Suggested enhancement for the future, we may want to look into pulling `faker` for generating some of these values.

Reply:
Good flag. I'll create an issue for it.